src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -37,8 +37,17 @@ def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
 
         col_identifier = str(column.identifier)
 
-        datum = data[col_identifier]
-        value = getattr(data_row, col_identifier)
+        # obtain data from 'col_identifier'
+        # and constructing if doesn't exists
+        try:
+            datum = data[col_identifier]
+        except KeyError:
+            data[col_identifier] = TogaData.alloc().init()
+            data_row._impl[col_identifier] = data[col_identifier]
+            datum = data[col_identifier]
```

Comment:
Although the diff doesn't show this well, we've now got two places where the TogaData instance is being created. Given that there's no real advantage to instantiating the table data upfront (as the table is drawn with separate calls for each column), I think it would be preferable to consolidate the logic so that we're *only* doing this on-demand population of TogaData instances. (i.e., lines 32-35 get deleted, and this try-except block is used for *all* attributes, not just the ones that originally existed.)

Reply:
I changed but I didn't remove the AttributeError because it's necessary for initialise data.